scripts/functions/diagnostic_application.sh

```diff
@@ -0,0 +1,114 @@
+#!/bin/bash
+#* This file is part of the MOOSE framework
+#* https://www.mooseframework.org
+#*
+#* All rights reserved, see COPYRIGHT for full restrictions
+#* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+#*
+#* Licensed under LGPL 2.1, please see LICENSE for details
+#* https://www.gnu.org/licenses/lgpl-2.1.html
+
+function enter_moose()
+{
+    # TODO: allow a --use-moose-dir argument
+    cd $CTMP_DIR/moose || exit 1
+}
+
+function clone_moose()
+{
+    # TODO: allow a --use-moose-dir argument
+    printf "Creating a clean clone of MOOSE repository\n"
+    git clone --depth 1 https://github.com/idaholab/moose ${CTMP_DIR}/moose -b master &> ${CTMP_DIR}/moose_clone_stdouterr.log
+    local exit_code=$?
+    if [ ${exit_code} -ge 1 ] && [ $(cat ${CTMP_DIR}/moose_clone_stdouterr.log | grep -c -i 'SSL certificate problem') -ge 1 ]; then
+        print_orange "WARNING: "
+        printf "SSL issues detected, attempting again with SSL protections off
+
+This may indicate the cause of other issues. PETSc contribs may fail to
+download properly.\n\n"
+        print_orange "export GIT_SSL_NO_VERIFY=true\n\n"
+        export GIT_SSL_NO_VERIFY=true
+        clone_moose
+        unset GIT_SSL_NO_VERIFY
+        return
+    fi
+    # Print relevant repo data
+    git -C ${CTMP_DIR}/moose rev-parse HEAD
+    git -C ${CTMP_DIR}/moose branch
+    git -C ${CTMP_DIR}/moose status
+}
+
+function build_library()
+{
+    if [ "${FULL_BUILD}" == 0 ]; then return; fi
+    print_sep
+    local error_cnt=${error_cnt:-0}
+    if [ ${error_cnt} -le 0 ]; then printf "Build Step: $1\n"; fi
+    enter_moose
+    if [ "$1" == 'petsc' ]; then
+        # PETSc is special due to all the contribs.
+        mkdir -p $CTMP_DIR/downloads
+        local petsc_urls=(`scripts/update_and_rebuild_petsc.sh --with-packages-download-dir=$CTMP_DIR/downloads 2>/dev/null | grep "\[" | cut -d, -f 2 | sed -e "s/\]//g"`)
+        cd $CTMP_DIR/downloads
+        printf "Downloading PETSc contribs...\n"
+        for petsc_url in "${petsc_urls[@]}"; do
+            clean_url=$(echo ${petsc_url} | sed -e "s/'//g")
+            curl --insecure -L -O $clean_url 2>/dev/null
```

Comment:
--insecure.   maybe only if GIT_SSL set to dangferous already?

Reply:
I should be able to get rid of all the insecure lines.

These are here because I didn't know all the influential environment variables needed to be passed into my pristine environment in order to do this securely.